### PR TITLE
Add Stack Diff Command

### PR DESCRIPTION
**Pull request by sp-joseluis-ledesma**
Although ChangeSets provides insightful information, sometimes it does not give enough information (for example when changing Instance Tags or PolicyDocuments). This diff command will show just a raw diff between the local generated template and the Cloudformation remote one.

-----------------

* [x] Wrote [good commit messages][1].
* [x] [Squashed related commits together][2].
* [ ] Commit message starts with `[Resolve #issue-number]` (if a related issue exists).
* [x] Added unit tests.
* [x] Added integration tests (if applicable).
* [x] All unit tests (`make test`) are passing.
* [x] Used the same coding conventions as the rest of the project.
* [x] The new code doesn't generate flake8 (`make lint`) offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.

### CODE REVIEW

File: sceptre/cli.py [hunk 1]
```diff
@@ -235,6 +235,22 @@ def describe_stack_resources(ctx, environment, stack):
...
+@cli.command(name="diff-stack")
+@stack_options
+@click.pass_context
+@catch_exceptions
+def diff_stack(ctx, environment, stack):
+    """
+    Shows the local template differences against cloudformation.
+
+    Prints differences.
+    """
```
Comment (b-t-g) [inline sceptre/cli.py]: This doc string should be more informative. Something to the extent of, "Print the diff between the currently deployed stack with the local version of that stack" would be better, in my opinion.

File: sceptre/cli.py [hunk 2]
```diff
@@ -235,6 +235,22 @@ def describe_stack_resources(ctx, environment, stack):
...
+    env = get_env(ctx.obj["sceptre_dir"], environment, ctx.obj["options"])
+    response = env.stacks[stack].diff()
+
+    write(response)
```
Comment (b-t-g) [inline sceptre/cli.py]: ```response``` should either be called ```diff``` or 

``` write(response)```
should be rewritten to be
```write(env.stacks[stack].diff())```
I have a weak preference for renaming "response" to "diff".

File: sceptre/stack.py [hunk 3]
```diff
@@ -384,6 +385,30 @@ def describe(self):
...
         )
 
+    def diff(self):
+        """
+        Returns the differences between the generated template and the
+        cloudformation one.
+
+        :returns: differences
+        :rtype: string
+        """
```
Comment (b-t-g) [inline sceptre/stack.py]: This docstring should be similarly updated.

File: tests/test_stack.py [hunk 4]
```diff
@@ -558,6 +558,20 @@ def test_describe_outputs_handles_stack_with_no_outputs(
...
+
+        self.stack.connection_manager.call.return_value = { "TemplateBody": "remote_template_content"}
+        response = self.stack.diff()
+        self.stack.connection_manager.call.assert_called_with(
+            service="cloudformation",
+            command="get_template",
+            kwargs={"StackName": sentinel.external_name}
+        )
+        print response
+        assert response == "--- remote\n+++ local\n@@ -1 +1 @@\n-remote_template_content\n+local_template_content\n"
```
Comment (b-t-g) [inline tests/test_stack.py]: I'd like to see more test cases. Adding a line before a given line, adding a line after a given line, adding a line in between two lines, as well as sanity checks when you switch which template you consider "remote" and which template you consider "local", the same lines still show up.

An example of what I mean by the sanity check:

Consider the following files:
foo with contents:
```
bar
```
bar with contents:
```
bar
baz
```
doing ```diff foo bar``` (the Unix diff, not a Sceptre diff) shows that baz was added. doing ```diff bar foo``` shows baz being removed. The Sceptre diff should behave similarly.

File: sceptre/stack.py [hunk 5]
```diff
@@ -384,6 +385,30 @@ def describe(self):
...
+        raw_remote = self.connection_manager.call(
+            service="cloudformation",
+            command="get_template",
+            kwargs={"StackName": self.external_name}
+        )["TemplateBody"]
+
+        raw_local = self.template.body
+
+        remote = raw_remote.split("\n")
+        local = raw_local.split("\n")
```
Comment (b-t-g) [inline sceptre/stack.py]: Could you make the variable names ```raw_remote```, ```raw_local``` (and their non-raw counterparts) more descriptive? Preferably something like ```raw_remote_template```, ```raw_local_template```, etc.

**Comment by sp-joseluis-ledesma** [comment]
Hi @b-t-g, I've make the requested changes.

File: sceptre/stack.py [hunk 6]
```diff
@@ -384,6 +385,30 @@ def describe(self):
...
+            kwargs={"StackName": self.external_name}
+        )["TemplateBody"]
+
+        raw_local_template = self.template.body
+
+        remote_template = raw_remote_template.split("\n")
+        local_template = raw_local_template.split("\n")
+        output = ""
+        for line in difflib.unified_diff(remote_template, local_template, fromfile="remote_template",
+                                         tofile="local_template", lineterm=""):
```
Comment (theseanything) [inline sceptre/stack.py]: set `difflib.unified_diff(remote_template, local_template, fromfile="remote_template", tofile="local_template", lineterm="")` as a variable on its own line.

File: sceptre/stack.py [hunk 7]
```diff
@@ -384,6 +385,30 @@ def describe(self):
             kwargs={"StackName": self.external_name}
         )
 
+    def diff(self):
+        """
+        Returns the diff between the currently deployed stack with the local
```
Comment (theseanything) [inline sceptre/stack.py]: ...between **the template body*** of the currently....

**Comment by sp-joseluis-ledesma** [comment]
I have made the changes requested. I have no experience with integration-tests :/

**Pull request merged**